execution/wasm/contract.go

```diff
@@ -220,13 +222,126 @@ func (ctx *context) ResolveFunc(module, field string) lifeExec.FunctionImport {
 	}
 
 	switch field {
+	case "create":
+		return func(vm *lifeExec.VirtualMachine) int64 {
+			valuePtr := int(uint32(vm.GetCurrentFrame().Locals[0]))
+			dataPtr := uint32(vm.GetCurrentFrame().Locals[1])
+			dataLen := uint32(vm.GetCurrentFrame().Locals[2])
+			resultPtr := uint32(vm.GetCurrentFrame().Locals[3])
+
+			// TODO: is this guaranteed to be okay? Should be avoid panic here if out of bounds?
+			value := bin.BigIntFromLittleEndianBytes(vm.Memory[valuePtr : valuePtr+ValueByteSize])
+
+			var data []byte
+			copy(data, vm.Memory[dataPtr:dataPtr+dataLen])
+
+			ctx.sequence++
+			nonce := make([]byte, txs.HashLength+8)
+			copy(nonce, ctx.vm.options.Nonce)
+			binary.BigEndian.PutUint64(nonce[txs.HashLength:], ctx.sequence)
+			newAccountAddress := crypto.NewContractAddress(ctx.params.Callee, nonce)
+
+			err := engine.EnsurePermission(ctx.state.CallFrame, ctx.params.Callee, permission.CreateContract)
+			if err != nil {
+				return Error
+			}
+
+			res, err := engine.CallFromSite(ctx.state, ctx.vm.externalDispatcher, ctx.params,
+				engine.CallParams{
+					Caller: ctx.params.Caller,
+					Callee: newAccountAddress,
+					Input:  vm.Memory[dataPtr : dataPtr+dataLen],
+					Value:  *value,
+					Gas:    ctx.params.Gas,
+				})
+
+			if err != nil {
+				if errors.GetCode(err) == errors.Codes.ExecutionReverted {
+					return Revert
+				}
+				panic(err)
+			}
+			err = engine.InitWASMCode(ctx.state, newAccountAddress, res)
+			if err != nil {
+				if errors.GetCode(err) == errors.Codes.ExecutionReverted {
+					return Revert
+				}
+				panic(err)
+			}
+
+			copy(vm.Memory[resultPtr:], res)
+
+			return Success
+		}
+
+	case "getBlockDifficulty":
+		return func(vm *lifeExec.VirtualMachine) int64 {
+			resultPtr := int(vm.GetCurrentFrame().Locals[0])
+
+			// set it to 1
+			binary.BigEndian.PutUint64(vm.Memory[resultPtr:resultPtr+32], 1)
+			return 0
+		}
+
+	case "getTxGasPrice":
+		return func(vm *lifeExec.VirtualMachine) int64 {
+			resultPtr := int(vm.GetCurrentFrame().Locals[0])
+
+			// set it to 1
+			binary.BigEndian.PutUint64(vm.Memory[resultPtr:resultPtr+16], 1)
+			return 0
```

Comment:
return Success; (also the other functions)

Reply:
Fixed!